internal/gatewayapi/helpers.go

```diff
@@ -305,19 +305,24 @@ func hostnameMatchesWildcardHostname(hostname, wildcardHostname string) bool {
 
 func containsPort(ports []*protocolPort, port *protocolPort) bool {
 	for _, protocolPort := range ports {
-		if protocolPort.port == port.port && layer4Protocol(protocolPort) == layer4Protocol(port) {
+		oldProtocol, oldLevel := layer4Protocol(protocolPort)
```

Comment:
curious why you chose to use `old` and `new` prefix here, doesnt seem to add up here
wouldn't `my` and `cur` be more apt ?